### PR TITLE
[Enhancement] Clear storage after lake table is dropped

### PR DESCRIPTION
**Pull request by sduzh**
Send `LakeService.dropTable` request to BE after lake table is dropped.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 69 / 88 (78.41%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[72, 73]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|3|7|42.86%|[381, 751, 752, 753]|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|1|2|50.00%|[66]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|10|70.00%|[2700, 2704, 2705]|
|:large_blue_circle:|com/starrocks/lake/ClearLakeTableTask.java|33|41|80.49%|[69, 70, 71, 72, 73, 74, 78, 79]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/ClearTableTaskList.java|14|14|100.00%|[]|

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.task;
+
+public interface ClearTableTask {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java:5]: compared to `runnable`, `ClearTableTask` uses `execute()` interface and can return a `boolean` value. However the return value is only checked in UT but not in production code. Can we say that `ClearTableTask` has no vital difference than `runnable`?

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java:5]: Yes, you're right.

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java:5]: lol.

So the next question is, whether it is necessary to have the `ClearTableTask` refactor? Can both `ClearOlapTableTask` and `ClearLakeTableTask` implements `Runnable` and archive the same goal?

File: fe/fe-core/src/main/java/com/starrocks/task/ClearTableTaskList.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.task;
+
+import com.clearspring.analytics.util.Lists;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/task/ClearTableTaskList.java:5]:
```suggestion
import com.google.common.collect.Lists;
```

File: fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/task/ClearTableTask.java:5]: Not really necessary now. Actually this patch was extracted from another patch, which already closed now, and the result value did used in that patch and I just keep that change here.

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 64 (26.56%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[81, 82]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableTask.java|4|40|10.00%|[35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 47, 48, 51, 52, 53, 54, 55, 57, 58, 59, 61, 63, 64, 68, 69, 70, 71, 72, 73, 77, 78, 80, 81, 82, 83]|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|1|2|50.00%|[66]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|15|53.33%|[467, 487, 488, 489, 2708, 2712, 2713]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|2|100.00%|[]|

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by sduzh** [comment]
The be unit test failed but I didn't change any BE code.

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Pull request merged**